### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by laurentpellegrino** [hcloud-talos/terraform-hcloud-talos#22] [General]
I am trying to create a fresh cluster using this project but I cannot succeed following the README.

I installed all recommended tools (using Terraform 1.8.4).

Images have been created with success using packer (note there is a permission issue on the script when cloning the Github repository and the script _create.sh_ must be invoked from within the folder _\_packer_ unlike said).

Then, I create a main.tf with a definition for the "talos" module using my own values and I appended the following as mentioned in the README:

```
output "talosconfig" {
  value     = module.talos.talosconfig
  sensitive = true
}

output "kubeconfig" {
  value     = module.talos.kubeconfig
  sensitive = true
}
```

When I run `terraform init` I get this error:

> Error: Duplicate output definition

If I remove the output definitions, `terraform init` completes. However, `terraform apply` requests input for cluster name, etc. as if my main.tf who contains all this information was ignored.

How to avoid this and make terraform use my _main.tf_?

**Comment by laurentpellegrino** [comment]
Note that to make `terraform init` work, I had to edit version in .terraform/modules/talos/terraform.tf 

Otherwise I get this error:

> │ Error: Unsupported Terraform Core version
> │ 
> │   on .terraform/modules/talos/terraform.tf line 2, in terraform:
> │    2:   required_version = "1.7.5"

**Reply by mrclrchtr** [reply]
this is not the talos version, it's the required terraform version. I'm not sure, why it's `1.7.5` in your case.
In this module it has to be `>= 1.7.0`
https://github.com/hcloud-talos/terraform-hcloud-talos/blob/f9b55bf6cb12d8fa311f8978bf821d58787a4721/terraform.tf#L2

**Comment by laurentpellegrino** [comment]
With help of ChatGPT, instead of creating a main.tf, I created a terraform.tfvars:

> hcloud_token            = "your-hcloud-token"
> cluster_name            = "dummy.com"
> cluster_domain          = "cluster.dummy.com.local"
> cluster_api_host        = "kube.dummy.com"
> firewall_use_current_ip = true
> datacenter_name         = "fsn1-dc14"
> control_plane_count     = 3
> control_plane_server_type = "cax11"
> talos_version           = "v1.7.2"
> worker_count            = 3
> worker_server_type      = "cax21"


Note the version for talos must start with _v_!

**Comment by laurentpellegrino** [comment]
Then, `terraform apply` goes farther:

> │ Error: Output refers to sensitive values
> │ 
> │   on outputs.tf line 11:
> │   11: output "talos_client_configuration" {
> │ 
> │ To reduce the risk of accidentally exporting sensitive data that was intended to be only internal, Terraform requires that any root module output containing sensitive data be explicitly
> │ marked as sensitive, to confirm your intent.
> │ 
> │ If you do intend to export this data, annotate the output value as sensitive by adding the following argument:
> │     sensitive = true
>

**Comment by laurentpellegrino** [comment]
If I edit _outputs.tf_ to add _sensitive = true_. 

Then, `terraform apply` generates the plan but fails on applying it since it was partially applied on previous runs and subsequent runs are not handling the case where resource are already defined:

> │ Error: name is already used (uniqueness_error)
> │ 
> │   with hcloud_firewall.this,
> │   on firewall.tf line 67, in resource "hcloud_firewall" "this":
> │   67: resource "hcloud_firewall" "this" {
> │ 
> ╵
> ╷
> │ Error: name is already used (uniqueness_error)
> │ 
> │   with hcloud_network.this,
> │   on network.tf line 10, in resource "hcloud_network" "this":
> │   10: resource "hcloud_network" "this" {
> │ 
> ╵
> ╷
> │ Error: name is already used (uniqueness_error)
> │ 
> │   with hcloud_primary_ip.worker_ipv4[2],
> │   on network.tf line 68, in resource "hcloud_primary_ip" "worker_ipv4":
> │   68: resource "hcloud_primary_ip" "worker_ipv4" {
> │ 
> ╵
> ╷
> │ Error: name is already used (uniqueness_error)
> │ 
> │   with hcloud_placement_group.worker,
> │   on placement_groups.tf line 6, in resource "hcloud_placement_group" "worker":
> │    6: resource "hcloud_placement_group" "worker" {
> │ 
> ╵
> ╷
> │ Error: API request failed
> │ 
> │   with hcloud_ssh_key.this,
> │   on server.tf line 38, in resource "hcloud_ssh_key" "this":
> │   38: resource "hcloud_ssh_key" "this" {
> │ 
> │ An unexpected error was encountered during an API request.
> │ 
> │ SSH key name is already used
> │ 
> │ Error code: uniqueness_error
> │ Status code: 409
> 
> 

For somebody new to the project with no knowledge about terraform, the README is not helping at all.

**Reply by mrclrchtr** [reply]
That is correct, this project here is in the absolute alpha phase and certainly not intended to explain the basics of terraform. I would recommend first understanding terraform and then using this module.

**Reply by laurentpellegrino** [reply]
Thanks. I understand. Unfortunately, even with Terraform knowledge, it seems several things are broken or not kept in a working state.

**Reply by laurentpellegrino** [reply]
Also note the version number used for releases is misleading for an alpha.

**Reply by mrclrchtr** [reply]
It's semantic release versioning: https://semver.org 

>  eliminating emotion from version numbers

**Reply by laurentpellegrino** [reply]
> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. 

> Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92, 1.0.0-x-y-z.--.

**Comment by mrclrchtr** [comment]
I am willing to help you, but please try to use threads and please try to ask clear questions.
